user_install.sh

```diff
@@ -0,0 +1,404 @@
+
+if !($OS_CHECK); then
+
+	# The script will not allow the user to install on an unsupported OS
+	instwarn "Terminating installation due to incorrect OS (detected $DETECTED_OS_CODENAME)"
+	instwarn "This project requires $REQUIRED_OS_RELEASE $REQUIRED_OS_RELEASE ($REQUIRED_OS_CODENAME)"
+	exit 1
+fi
+
+if !($ROOT_CHECK); then
+
+	# The script will not allow the user to install as root
+	instwarn "Terminating installation due to forbidden user account"
+	instwarn "The install script should not be run as root"
+	exit 1
+fi
+
+if !($BASHRC_CHECK); then
+
+	# The script will not allow the user to install if ROS is being sourced
+	instwarn "Terminating installation due to $BASHRC_FILE sourcing a ROS version"
+	instwarn "This should be handled through the MIL runcom file instead"
+	instwarn "Removing lines that source ROS, workspaces, or project aliases is recommended"
+	instwarn "However, removing only lines that source ROS will be enough to clear this error"
+	exit 1
+fi
+
+
+#=================================#
+# Workspace and Repository Set Up #
+#=================================#
+
+# Source ROS configurations for bash
+source /opt/ros/$ROS_VERSION/setup.bash
+
+# Set up catkin workspace directory
+if !([ -f $CATKIN_DIR/src/CMakeLists.txt ]); then
+	instlog "Generating catkin workspace at $CATKIN_DIR"
+	mkdir -p $CATKIN_DIR/src
+	cd $CATKIN_DIR/src
+	catkin_init_workspace
+	catkin_make -C $CATKIN_DIR -B
+else
+	instlog "Using existing catkin workspace at $CATKIN_DIR"
+fi
+
+# Source the workspace's configurations for bash
+source $CATKIN_DIR/devel/setup.bash
+
+# Download the software-common repository if it has not already been downloaded
+if !(ls $CATKIN_DIR/src | grep --quiet "software-common"); then
+	instlog "Downloading the software-common repository"
+	cd $CATKIN_DIR/src
+	git clone --recursive -q https://github.com/uf-mil/software-common.git
+	cd $CATKIN_DIR/src/software-common
+	git remote rename origin upstream
+	if [ ! -z "$SWC_USER_FORK" ]; then
+		git remote add origin "$SWC_USER_FORK"
+	fi
+fi
+
+# Download the Sub8 repository if it has not already been downloaded and was selected for installation
+if ($INSTALL_SUB) && !(ls $CATKIN_DIR/src | grep --quiet "Sub8"); then
+	instlog "Downloading the Sub8 repository"
+	cd $CATKIN_DIR/src
+	git clone --recursive -q https://github.com/uf-mil/Sub8.git
+	cd $CATKIN_DIR/src/Sub8
+	git remote rename origin upstream
+	if [ ! -z "$SUB_USER_FORK" ]; then
+		git remote add origin "$SUB_USER_FORK"
+	fi
+fi
+
+# Download the Navigator repository if it has not already been downloaded and was selected for installation
+if ($INSTALL_NAV); then
+	if !(ls $CATKIN_DIR/src | grep --quiet "Sub8"); then
+		instlog "Downloading the Sub8 repository"
+		cd $CATKIN_DIR/src
+		git clone --recursive -q https://github.com/uf-mil/Sub8.git
+		cd $CATKIN_DIR/src/Sub8
+		instlog "Rolling back the Sub8 repository; do not pull the latest version!"
+		git reset --hard 0089e68b9f48b96af9c3821f356e3a487841e87e
```

Comment:
NaviGator still depends on the Sub8 repository for some reason. If you're installing NaviGator, the script pulls Sub8 and resets it to the last commit that was tested with that platform. This won't affect pure SubjuGator installations, though.